src/components/TextStack/TextStack.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC } from 'react';
+
+import classNames from 'classnames';
+import { AsyncText } from '../../lib/types';
+
+import { Skeleton } from '../Skeleton';
+
+import styles from './TextStack.module.scss';
+
+export interface TextStackProps {
+  className?: string;
+  label: string;
+  primaryText: AsyncText | string;
+  secondaryText: AsyncText | string;
+}
+
+const SkeletonText = ({ text }: { text: AsyncText | string }) => {
+  let el;
+  if (typeof text === 'object') {
+    // If loading return skeleton, else return text (or ERR if undefined)
+    el = text.isLoading ? <Skeleton /> : text.text ?? 'ERR';
+  } else {
+    el = text;
+  }
+  return <>{el}</>;
```

Comment:
I think it is the same 

```
if (typeof text === 'object') {
    // If loading return skeleton, else return text (or ERR if undefined)
    return text.isLoading ? <Skeleton /> : text.text ?? 'ERR';
  } 
  return text;
```

And `const SkeletonText = ({ text }: { text: AsyncText | string }): ReactNode => {`